api/api-job-instance-v1/src/main/java/com/thoughtworks/go/apiv1/jobinstance/representers/JobInstanceRepresenter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.jobinstance.representers;
+
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.domain.JobInstance;
+
+import static java.util.Collections.emptyList;
+
+public class JobInstanceRepresenter {
+    public static void toJSON(OutputWriter outputWriter, JobInstance jobInstance) {
+        outputWriter.add("name", jobInstance.getName());
+        if (jobInstance.getState() != null) {
+            outputWriter.add("state", jobInstance.getState().toString());
+        }
+        if (jobInstance.getResult() != null) {
+            outputWriter.add("result", jobInstance.getResult().toString());
+        }
+        if (jobInstance.isNull()) {
+            renderNullJobInstance(outputWriter);
+            return;
+        }
```

Comment:
How about returning a 404 instead of 200 for a `NullJobInstance`.

As we're bumping up the API version, we could consider this as bug-fix.

Reply:
Agreed. Throwing a `RecordNotFoundException` in case a `NullJobInstance` is received from the service.